content/types/_channel_event.textile

```diff
@@ -1,25 +1,92 @@
-@ARTChannelEvent@ is the enum emitted as the event in @ARTRealtimeChannel.on@. The variants correspond to "@Realtime Channel@ state":/realtime/channels-messages/#channel-state changes, plus an arbitrary @Error@.
-
-```[objc]
-  typedef NS_ENUM(NSUInteger, ARTChannelEvent) {
-      ARTChannelEventInitialized,
-      ARTChannelEventAttaching,
-      ARTChannelEventAttached,
-      ARTChannelEventDetaching,
-      ARTChannelEventDetached,
-      ARTChannelEventFailed,
-      ARTChannelEventError
-  };
-```
-
-```[swift]
-  enum ARTChannelEvent : UInt {
-      case Initialized
-      case Attaching
-      case Attached
-      case Detaching
-      case Detached
-      case Failed
-      case Error
-  }
-```
+blang[jsall].
+  @ChannelEvent@ is a String that can be emitted as an event on the @Channel@ object; either a "@ChannelState@":/realtime/channels-messages#channel-states or an @update@ event.
+
+
+  ```[javascript]
+    var allChannelEvents = allChannelStates.concat(['update']);
```

Comment:
I am not sure I understand what you are demonstrating here given `allChannelStates` is not defined.  Could you rather not just list out all possible string values like we do for other enums making it much clearer to users what to expect?